examples/dcutr/src/main.rs

```diff
@@ -155,21 +122,34 @@ fn main() -> Result<(), Box<dyn Error>> {
         }
     }
 
-    let behaviour = Behaviour {
-        relay_client: client,
-        ping: ping::Behaviour::new(ping::Config::new()),
-        identify: identify::Behaviour::new(identify::Config::new(
-            "/TODO/0.0.1".to_string(),
-            local_key.public(),
-        )),
-        dcutr: dcutr::Behaviour::new(local_peer_id),
-    };
+    let mut swarm = libp2p::builder::SwarmBuilder::new()
+        .with_existing_identity(generate_ed25519(opts.secret_key_seed))
+        .with_async_std()
+        .with_tcp_config(tcp::Config::default().port_reuse(true).nodelay(true))
+        .with_noise()
+        .with_relay()
+        .with_noise()
+        .with_other_transport(|keypair| {
+            quic::async_std::Transport::new(quic::Config::new(&keypair))
+                .map(|(peer_id, muxer), _| (peer_id, StreamMuxerBox::new(muxer)))
+        })
+        .without_any_other_transports()
+        .with_dns()
+        .await
+        // TODO: shortcut
+        .without_websocket()
+        .with_behaviour(|keypair, relay_behaviour| Behaviour {
+            relay_client: relay_behaviour,
+            ping: ping::Behaviour::new(ping::Config::new()),
+            identify: identify::Behaviour::new(identify::Config::new(
+                "/TODO/0.0.1".to_string(),
+                keypair.public(),
+            )),
+            dcutr: dcutr::Behaviour::new(keypair.public().to_peer_id()),
+        })
+        .build();
 
-    let mut swarm = match ThreadPool::new() {
-        Ok(tp) => SwarmBuilder::with_executor(transport, behaviour, local_peer_id, tp),
-        Err(_) => SwarmBuilder::without_executor(transport, behaviour, local_peer_id),
-    }
-    .build();
+    info!("Local peer id: {:?}", swarm.local_peer_id());
```

Comment:
Worth considering adding a generic log to the boot of a `Swarm` I think? In separate PR though :)

Reply:
Good idea. Tracked in https://github.com/libp2p/rust-libp2p/issues/4394.